components/motor/ulnstepper/28byj-48.go

```diff
@@ -0,0 +1,381 @@
+			28byj-48:	https://components101.com/sites/default/files/component_datasheet/28byj48-step-motor-datasheet.pdf
+
+	This driver will drive the motor with half-step driving method (instead of full-step drive) for higher resolutions.
+	In half-step the current vector divides a circle into eight parts. The eight step switching sequence is shown in
+	stepSequence below. The motor takes 5.625*(1/64)° per step. For 360° the motor will take 4096 steps.
+
+	We set the minimum sleep time between steps to be 0.002s to prevent hardware damage. The motor also has a max speed
+	of 10-15 rpm at 5V.
+*/
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/board"
+	"go.viam.com/rdk/components/generic"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/config"
+	"go.viam.com/rdk/operation"
+	"go.viam.com/rdk/registry"
+	"go.viam.com/rdk/resource"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var (
+	model                = resource.NewDefaultModel("28byj48")
+	minDelayBetweenTicks = 100 * time.Microsecond // minimum sleep time between each ticks
+)
+
+// stepSequence contains switching signal for uln2003 pins.
+// Treversing through stepSequence once is one step.
+var stepSequence = [8][4]bool{
+	{false, false, false, true},
+	{true, false, false, true},
+	{true, false, false, false},
+	{true, true, false, false},
+	{false, true, false, false},
+	{false, true, true, false},
+	{false, false, true, false},
+	{false, false, true, true},
+}
+
+// PinConfig defines the mapping of where motor are wired.
+type PinConfig struct {
+	In1 string `json:"in1"`
+	In2 string `json:"in2"`
+	In3 string `json:"in3"`
+	In4 string `json:"in4"`
+}
+
+// Config describes the configuration of a motor.
+type Config struct {
+	Pins             PinConfig `json:"pins"`
+	BoardName        string    `json:"board"`
+	TicksPerRotation int       `json:"ticks_per_rotation"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (config *Config) Validate(path string) ([]string, error) {
+	var deps []string
+	if config.BoardName == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "board")
+	}
+
+	if config.Pins.In1 == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "In1")
+	}
+
+	if config.Pins.In2 == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "In2")
+	}
+
+	if config.Pins.In3 == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "In3")
+	}
+
+	if config.Pins.In4 == "" {
+		return nil, utils.NewConfigValidationFieldRequiredError(path, "In4")
+	}
+
+	deps = append(deps, config.BoardName)
+	return deps, nil
+}
+
+func init() {
+	_motor := registry.Component{
+		Constructor: func(ctx context.Context, deps registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
+			return new28byj(deps, config, config.Name, logger)
```

Comment:
Let's clean up the constructor so we can get to a:
`Constructor: new28byj` state